tools/api-markdown-documenter/src/api-item-transforms/configuration/Transformations.ts

```diff
@@ -55,138 +57,74 @@ export type TransformApiItemWithoutChildren<TApiItem extends ApiItem> = (
-
-	/**
-	 * Transformation to generate a {@link SectionNode} for a `Type Alias`.
-	 */
-	readonly transformApiTypeAlias: TransformApiItemWithoutChildren<ApiTypeAlias>;
-
-	/**
-	 * Transformation to generate a {@link SectionNode} for an `Variable`.
-	 */
-	readonly transformApiVariable: TransformApiItemWithoutChildren<ApiVariable>;
+	readonly [ApiItemKind.Model]: TransformApiItemWithoutChildren<ApiModel>;
+	readonly [ApiItemKind.Namespace]: TransformApiItemWithChildren<ApiNamespace>;
+	readonly [ApiItemKind.Property]: TransformApiItemWithoutChildren<ApiProperty>;
+	readonly [ApiItemKind.PropertySignature]: TransformApiItemWithoutChildren<ApiPropertySignature>;
+	readonly [ApiItemKind.TypeAlias]: TransformApiItemWithoutChildren<ApiTypeAlias>;
+	readonly [ApiItemKind.Variable]: TransformApiItemWithoutChildren<ApiVariable>;
 }
 
 /**
  * The default {@link ApiItemTransformationConfiguration}.
  */
 const defaultApiItemTransformationOptions: ApiItemTransformations = {
-	transformApiCallSignature: DefaultTransformationImplementations.transformApiItemWithoutChildren,
-	transformApiClass: DefaultTransformationImplementations.transformApiClass,
-	transformApiConstructor: DefaultTransformationImplementations.transformApiFunctionLike,
-	transformApiEntryPoint: DefaultTransformationImplementations.transformApiEntryPoint,
-	transformApiEnum: DefaultTransformationImplementations.transformApiEnum,
-	transformApiEnumMember: DefaultTransformationImplementations.transformApiItemWithoutChildren,
-	transformApiFunction: DefaultTransformationImplementations.transformApiFunctionLike,
-	transformApiIndexSignature:
+	[ApiItemKind.CallSignature]:
+		DefaultTransformationImplementations.transformApiItemWithoutChildren,
+	[ApiItemKind.Class]: DefaultTransformationImplementations.transformApiClass,
+	[ApiItemKind.Constructor]: DefaultTransformationImplementations.transformApiFunctionLike,
+	[ApiItemKind.ConstructSignature]: DefaultTransformationImplementations.transformApiFunctionLike,
+	[ApiItemKind.EntryPoint]: DefaultTransformationImplementations.transformApiEntryPoint,
+	[ApiItemKind.Enum]: DefaultTransformationImplementations.transformApiEnum,
+	[ApiItemKind.EnumMember]: DefaultTransformationImplementations.transformApiItemWithoutChildren,
+	[ApiItemKind.Function]: DefaultTransformationImplementations.transformApiFunctionLike,
+	[ApiItemKind.IndexSignature]:
+		DefaultTransformationImplementations.transformApiItemWithoutChildren,
+	[ApiItemKind.Interface]: DefaultTransformationImplementations.transformApiInterface,
+	[ApiItemKind.Method]: DefaultTransformationImplementations.transformApiFunctionLike,
+	[ApiItemKind.MethodSignature]: DefaultTransformationImplementations.transformApiFunctionLike,
+	[ApiItemKind.Model]: DefaultTransformationImplementations.transformApiModel,
+	[ApiItemKind.Namespace]: DefaultTransformationImplementations.transformApiNamespace,
+	// TODO: Make package transformation configurable
```

Comment:
"Package transformations" are an entry not present today that would go here alphabetically, right? Maybe clarify, I first got confused thinking this applied to the `Property` transform in the next line.

Reply:
Yeah, that's correct. And I can definitely see why this was confusing. I'll move this up to a `privateRemarks` comment on the type itself.